packages/make-styles-webpack-loader/src/webpackLoader.ts

```diff
@@ -0,0 +1,89 @@
+import { EvalCache, Module } from '@linaria/babel';
+import * as enhancedResolve from 'enhanced-resolve';
+import * as path from 'path';
+import * as webpack from 'webpack';
+
+import { transformSync, TransformResult, TransformOptions } from './transformSync';
+
+type WebpackLoaderParams = Parameters<webpack.LoaderDefinitionFunction<never>>;
+
+/**
+ * Webpack can also pass sourcemaps as a string, Babel accepts only objects.
+ * See https://github.com/babel/babel-loader/pull/889.
+ */
+function parseSourceMap(inputSourceMap: WebpackLoaderParams[1]): TransformOptions['inputSourceMap'] {
+  try {
+    if (typeof inputSourceMap === 'string') {
+      return JSON.parse(inputSourceMap) as TransformOptions['inputSourceMap'];
+    }
+
+    return inputSourceMap as TransformOptions['inputSourceMap'];
+  } catch (err) {
+    return undefined;
+  }
+}
+
+export function webpackLoader(
+  this: webpack.LoaderContext<never>,
+  sourceCode: WebpackLoaderParams[0],
+  inputSourceMap: WebpackLoaderParams[1],
+) {
+  EvalCache.clearForFile(this.resourcePath);
+
+  const resolveOptionsDefaults: webpack.ResolveOptions = {
+    conditionNames: ['require'],
+    extensions: ['.js', '.jsx', '.ts', '.tsx', '.json'],
+  };
+  // ⚠ "this._compilation" is a deprecated API, however there seems to be no other way to access Webpack's resolver
```

Comment:
current docs don't state that `this._compilation` is a deprecated API, just that using it will reduce the portability of the loader:

https://webpack.js.org/api/loaders/

Reply:
Good catch, this was stated in Linaria's code. Modified the comment 👍